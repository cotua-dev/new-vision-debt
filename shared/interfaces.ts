export interface PageProps {
    props: {};
    revalidate: number;
};
