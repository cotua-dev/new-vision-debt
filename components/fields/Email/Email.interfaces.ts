import { Dispatch, SetStateAction } from 'react';
import { Step } from '../../Stepper/Stepper.interfaces';

export interface EmailProps {
    steps: Step[];
    currentStep: number;
    setDisableNextButton: Dispatch<SetStateAction<boolean>>;
    email: string;
    setEmail: Dispatch<SetStateAction<string>>;
};
