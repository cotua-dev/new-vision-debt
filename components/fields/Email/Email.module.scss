@use 'styles/mixins';

div.email-field-wrapper {
    @include mixins.flexify(center, center, column);

    input.email-field {
        @include mixins.base-field;
    }

    small.error {
        @include mixins.error;
    }
}
