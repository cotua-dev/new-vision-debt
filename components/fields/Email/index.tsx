import { ChangeEvent, useState } from 'react';
import { EmailProps } from './Email.interfaces';
import styles from './Email.module.scss';

export function Email(props: EmailProps): JSX.Element {
    const { steps, currentStep, setDisableNextButton, email, setEmail } = props;

    // Cite: https://www.emailregex.com/
    const emailRegex: RegExp = new RegExp(/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/);

    const [error, setError] = useState('');
    const [showError, setShowError] = useState(false);

    /**
     * Set the value of email for the step and `email`
     * @param value String of the email
     */
    function handleEmailSetting(value: string): void {
        // Set `email`
        setEmail(value);

        // Check to see if the value passes the regular expression test
        if (emailRegex.test(value)) {
            // Set and remove error
            setError('');
            setShowError(false);

            // Set the current step's `email` value
            steps[currentStep].value.email = value;

            // Make the current step valid
            steps[currentStep].validity = true;

            // Enable the next button
            setDisableNextButton(false);
        } else {
            // Set and show error
            setError('Please provide an email');
            setShowError(true);

            // Disable the next button
            setDisableNextButton(true);
        }
    }

    return (
        <div className={styles['email-field-wrapper']}>
            <input
                className={styles['email-field']}
                type="email"
                id="email"
                name="email"
                onChange={(event: ChangeEvent<HTMLInputElement>) => handleEmailSetting(event.target.value)}
                value={email}
            />
            {
                showError &&
                <small className={styles['error']}>{error}</small>
            }
        </div>
    );
}
