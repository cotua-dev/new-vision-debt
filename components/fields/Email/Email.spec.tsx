import { render, RenderResult } from '@testing-library/react';
import { EmailProps } from './Email.interfaces';
import { Email } from './index';

describe('Email Component', (): void => {
    let renderResult: RenderResult;
    const props: EmailProps = {
        steps: [],
        currentStep: 0,
        setDisableNextButton: () => {},
        email: '',
        setEmail: () => {},
    };

    beforeAll((): RenderResult => renderResult = render(<Email {...props}/>));

    test('renders', (): void => {
        const { container } = renderResult;
        expect(container.firstChild).toMatchSnapshot();
    });

    afterAll((): void => renderResult.unmount());
});
