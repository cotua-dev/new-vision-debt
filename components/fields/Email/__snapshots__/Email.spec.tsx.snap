// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Component renders 1`] = `
<div
  class="email-field-wrapper"
>
  <input
    class="email-field"
    id="email"
    name="email"
    type="email"
    value=""
  />
</div>
`;
